src/main/java/frc/robot/commands/shooter/AdvancedSpinningShot.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.commands.shooter;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Shooter;
+import frc.robot.utils.ShooterSpeed;
+
+import java.util.function.Supplier;
+
+public class AdvancedSpinningShot extends Command {
+    private final Shooter shooter;
+    private final Supplier<Pose2d> pose2dSupplier;
+    private final Timer timer;
+    private ShooterSpeed shooterSpeed;
+
+    public AdvancedSpinningShot(Shooter shooter, Supplier<Pose2d> curentPoseSupplier) {
+        this.shooter = shooter;
+        this.pose2dSupplier = curentPoseSupplier;
+        this.timer = new Timer();
+    }
+
+    @Override
+    public void initialize() {
+        timer.reset();
+        this.shooterSpeed = calcuateShooterSpeed();
+        timer.start();
+    }
+
+    @Override
+    public void execute() {
+        shooter.setShooterMotorLeftSpeed(shooterSpeed.getLeftMotorSpeed());
+        shooter.setShooterMotorRightSpeed(shooterSpeed.getRightMotorSpeed());
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        shooter.stopShooter();
+        timer.stop();
+    }
+
+    @Override
+    public boolean isFinished() {
+        return timer.hasElapsed(Constants.ADVANCED_SPINNING_SHOT_TIMEOUT);
+    }
+
+    private ShooterSpeed calcuateShooterSpeed() {
```

Comment:
Is this trying to flip the motor speeds based on what side of the speaker we're on? @Sahiltheram  has done similar work. Can you two work together to determine what is needed>

Reply:
This code does a slightly different thing and will be used if we can get a higher level of automation. This code mitigates the shooting spin oscillation problem and should only be used in tangent with the auto steer code. Bat-Chen, Ohad, and I discussed this yesterday, and I would be happy to give you a more in-depth explanation in person.